python/taichi/lang/impl.py

```diff
@@ -607,20 +607,6 @@ def ndarray(dtype, shape):
     return ScalarNdarray(dtype, shape)
 
 
-class Layout:
-    def __init__(self, soa=False):
-        self.soa = soa
-
-
-SOA = Layout(soa=True)
```

Comment:
Let's not deprecate this yet, or it would break users. Could you annotate this with a deprecated warning?

Reply:
Do you mean we still keep this class for use? Or should I directly refer `ti.SOA` to `ti.Layout.SOA`? (But in this case I don't know how to give a deprecated warning)

Reply:
This can be a fancy way to achieve that: https://stackoverflow.com/questions/922550/how-to-mark-a-global-as-deprecated-in-python

Reply:
Wow interesting. However

> The downside is that you are now performing two lookups when you access the module, so there is a slight performance hit.

Can we afford this?

Reply:
Should be good :-)

Reply:
A problem arises - If we wrap `impl` as such class, then we can no longer use `from taichi.lang.impl import *` which is currently everywhere now. And we cannot expose things inside as `ti.xxx` either.

Reply:
OK, let's just make `AOS = Layout.AOS` in `__init__.py` for now

Reply:
Well I solved it just now - there's a new feature in Python 3.7 which allows us to directly add `__getattr__` to modules. Now I added a general way to handle deprecated names - directly add the mapping to `deprecated_names` in `taichi/__init__.py`.

Reply:
We support 3.6....

Reply:
Oh... Maybe directly do `AOS = Layout.AOS` for 3.6?

Reply:
Sounds good